nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/NviCandidateRepository.java

```diff
@@ -2,93 +2,159 @@
 import software.amazon.awssdk.enhanced.dynamodb.Key;
 import software.amazon.awssdk.enhanced.dynamodb.model.Page;
 import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional;
-import software.amazon.awssdk.enhanced.dynamodb.model.QueryEnhancedRequest;
 import software.amazon.awssdk.enhanced.dynamodb.model.TransactPutItemEnhancedRequest;
 import software.amazon.awssdk.enhanced.dynamodb.model.TransactWriteItemsEnhancedRequest;
+import software.amazon.awssdk.enhanced.dynamodb.model.TransactWriteItemsEnhancedRequest.Builder;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
-public class NviCandidateRepository extends DynamoRepository  {
+public class NviCandidateRepository extends DynamoRepository {
+
     private final DynamoDbTable<CandidateDao> candidateTable;
     private final DynamoDbTable<CandidateUniquenessEntry> uniquenessTable;
     private final DynamoDbIndex<CandidateDao> publicationIdIndex;
+    private final DynamoDbTable<ApprovalStatusDao> approvalStatusTable;
 
     public NviCandidateRepository(DynamoDbClient client) {
         super(client);
-        this.candidateTable = this.client.table(NVI_TABLE_NAME, CandidateDao.TABLE_SCHEMA);
-        this.uniquenessTable = this.client.table(NVI_TABLE_NAME, CandidateUniquenessEntry.TABLE_SCHEMA);
+        this.candidateTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(CandidateDao.class));
+        this.uniquenessTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(CandidateUniquenessEntry.class));
         this.publicationIdIndex = this.candidateTable.index(SECONDARY_INDEX_PUBLICATION_ID);
+        this.approvalStatusTable = this.client.table(NVI_TABLE_NAME, fromImmutableClass(ApprovalStatusDao.class));
     }
 
+    public Candidate create(DbCandidate dbCandidate, List<DbApprovalStatus> approvalStatuses) {
+        var identifier = UUID.randomUUID();
+        var candidate = new CandidateDao(identifier, dbCandidate);
+        var uniqueness = new CandidateUniquenessEntry(dbCandidate.publicationId().toString());
+        var transactionBuilder = buildTransaction(approvalStatuses, candidate, identifier, uniqueness);
 
-    public CandidateWithIdentifier create(Candidate candidate) {
-        var uuid = UUID.randomUUID();
-        var insert = new CandidateDao(uuid, candidate);
-        var uniqueness = new CandidateUniquenessEntry(candidate.publicationId().toString());
+        this.client.transactWriteItems(transactionBuilder.build());
+        var candidateObj = candidateTable.getItem(candidate);
 
-        var putCandidateRequest = TransactPutItemEnhancedRequest.builder(CandidateDao.class)
-                             .item(insert)
-                             .conditionExpression(uniquePrimaryKeysExpression())
-                             .build();
+        return new Candidate(candidateObj.identifier(), candidateObj.candidate(),
+                             getApprovalStatuses(approvalStatusTable, candidateObj.identifier()));
```

Comment:
Perhaps we can optimize this later, but now you do 2 queries to return candidate+approvalstatuses.
We could also have done it as 1 (everything with same PK0) fetched it as a generic Dao/DynamoEntry and filtered it into the 2 needed variables. This will be even more relevant when we implement Notes as its own rows

Reply:
I will not do it here and now, but read up on it, and prepare for when notes arrive.